modules/core/include/opencv2/core/cvdef.h

```diff
@@ -248,22 +248,18 @@ Cv64suf;
 #  define DISABLE_OPENCV_24_COMPATIBILITY
 #endif
 
-#if (defined _WIN32 || defined WINCE || defined __CYGWIN__) && defined CVAPI_EXPORTS
-#  define CV_EXPORTS __declspec(dllexport)
-#elif defined __GNUC__ && __GNUC__ >= 4
-#  define CV_EXPORTS __attribute__ ((visibility ("default")))
-#else
-#  define CV_EXPORTS
+#include "opencv2/cv_exports.h"
+
+// fix for Android FAT_JAVA_LIB: symbols should have default visibility even in static libraries
+#ifdef __ANDROID__
+#undef CV_EXPORTS
+#define CV_EXPORTS __attribute__ ((visibility ("default")))
 #endif
 
-#ifndef CV_DEPRECATED
-#  if defined(__GNUC__)
-#    define CV_DEPRECATED __attribute__ ((deprecated))
-#  elif defined(_MSC_VER)
-#    define CV_DEPRECATED __declspec(deprecated)
-#  else
-#    define CV_DEPRECATED
-#  endif
+#ifdef _MSC_VER
+#define CV_EXPORTS_TEMPLATE
+#else
+#define CV_EXPORTS_TEMPLATE CV_EXPORTS
```

Comment:
`CV_EXPORTS_TEMPLATE` is not expected by bindings parser.

Reply:
Currently bindings generators do not support templates, so we don't need it for now. We can easily add `CV_EXPORTS_TEMPLATE_W` macro if such functionality will be implemented.